pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1906,25 +1919,25 @@ impl Timeline {
 
         // Step 1: construct a k-merge iterator over all layers.
         // Also, verify if the layer map can be split by drawing a horizontal line at every LSN start/end split point.
-        let layer_names: Vec<crate::tenant::storage_layer::LayerName> = layer_selection
+        let layer_names = job_desc
+            .selected_layers
             .iter()
             .map(|layer| layer.layer_desc().layer_name())
             .collect_vec();
         if let Some(err) = check_valid_layermap(&layer_names) {
-            bail!("cannot run gc-compaction because {}", err);
+            warn!("gc-compaction layer map check failed because {}, this is normal if partial compaction is not finished yet", err);
         }
```

Comment:
I thought a week or two ago you said that you wanted to relax the definition of what a valid layer map is towards "as long as no layers overlap it's valid".

Reply:
Yes, but doing that check would be O(n^2) and I don't want to implement it that way, so let's warn here first.